packages/strapi-utils/lib/models.js

```diff
@@ -457,11 +457,25 @@ module.exports = {
     _.forEach(params, (value, key)  => {
       let result;
       let formattedValue;
-
-      // Check if the value if a valid candidate to be converted to a number value
-      formattedValue = isNumeric(value)
-        ? _.toNumber(value)
-        : value;
+      let modelAttributes = models[model]['attributes'];
+      let fieldType;
+      // Get the field type to later check if it's a string before number conversion
+      if (modelAttributes[key]) {
+        fieldType = modelAttributes[key]['type'];
+      } else {
+        let splitKey = key.split('_').pop();
```

Comment:
This returns the last element of the array. Let's take an example, in the case the received key equals to `title_contains`, it means that `let splitkey` will be equal to `contains` whereas we want the first element of the array.

```js
const splitKey = key.split('_')[0];
```

Reply:
If we follow my previous message and logic, we could simplify the algorithm like this

```js
const splitKey = key.split('_')[0];

if (modelAttributes[splitKey]) {
  fieldType = modelAttributes[splitKey]['type'];
}

if (fieldType !== 'string') {
  // ...
}
```

Reply:
Correct me if I'm wrong, but fields are allowed to have an underscore in them. I recall attempting to use something like initially, but if the field is called `underscore_field`, then only `underscore` is returned.

Reply:
Nevermind. I'm not sure how this one got past me. I could have sworn it was working correctly. Fix is on the way.